ambry-rest/src/main/java/com.github.ambry.rest/NettyServerChannelInitializer.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.rest;
+
+import com.github.ambry.config.NettyConfig;
+import com.github.ambry.network.SSLFactory;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.handler.stream.ChunkedWriteHandler;
+import io.netty.handler.timeout.IdleStateHandler;
+import java.net.InetSocketAddress;
+
+
+/**
+ * A {@link ChannelInitializer} to be used with {@link NettyServer}.
+ */
+public class NettyServerChannelInitializer extends ChannelInitializer<SocketChannel> {
```

Comment:
This can be package private. The reason it's a separate class is because I wanted to use it in `NettyServerTest` instead of duplicating this code. Do you think it will be better as an inner class?